vignettes/fims-path-maturity.Rmd

```diff
@@ -0,0 +1,614 @@
+
+```{r, class.source = "rchunk", eval = FALSE}
+# install and load FIMS
+remotes::install_github("NOAA-FIMS/FIMS@integration-test-estimation-R-clear-FIMS")
+library(FIMS)
+# Create a new maturity model
+fims <- Rcpp::Module("fims", PACKAGE = "FIMS")
+maturity <- new(fims$LogisticMaturity)
+# Populate the maturity module with parameter values.
+maturity$median$value <- 10
+maturity$median$is_random_effect <- FALSE
+maturity$median$estimated <- FALSE
+maturity$slope$value <- 0.2
+maturity$slope$is_random_effect <- FALSE
+maturity$slope$estimated <- FALSE
+```
+
+After that, we need to create and set up a population and link the maturity module to the population:
+```{r,  class.source = "rchunk", eval = FALSE}
+population <- new(fims$Population)
+#set up population specific parameters (other modules)
+#...
+#set maturity for the population - will throw error until PR #363 is merged into main
+population$SetMaturity(maturity$get_id())
+```
+
+Because FIMS sets up objects stored in memory, simply running `rm(fims)`, won't free this memory back up. There is a function that can be run to clear out memory:
+```{r,  class.source = "rchunk", eval = FALSE}
+# Clear C++ memory
+fims$clear()
+#Clear R memory
+rm(fims)
+rm(maturity)
+rm(population)
+```
+Also restarting or closing out your R session will free up memory.
+
+## The Rcpp Interface
+
+The fields specific to the maturity module that are accessible from R are defined in the 
+[rcpp_interface.hpp](https://github.com/NOAA-FIMS/FIMS/blob/main/inst/include/interface/rcpp/rcpp_interface.hpp) file under [inst/include/interface/rcpp](https://github.com/NOAA-FIMS/FIMS/tree/main/inst/include/interface/rcpp):
+
+```{Rcpp, class.source = "rcppchunk", eval = FALSE}
+Rcpp::class_<LogisticMaturityInterface>("LogisticMaturity")
+      .constructor()
+      .field("median", &LogisticMaturityInterface::median)
+      .field("slope", &LogisticMaturityInterface::slope)
+      .method("get_id", &LogisticMaturityInterface::get_id)
+      .method("evaluate", &LogisticMaturityInterface::evaluate);
+```
+
+The code above shows that there are two maturity module-specific fields, `median` and `slope`, and two maturity module-specific methods (i.e. functions), `get_id` and `evaluate`. You can access the fields and methods from R using the `show(fims)` function.
+
+The maturity module's `get_id` method returns a unique ID for this specific module.
+
+The maturity module is called from R using the name in quotes from the `Rcpp::class` function, `LogisticMaturity` and references the maturity rcpp class, `LogisticMaturityInterface` defined in [rcpp_maturity.hpp](https://github.com/NOAA-FIMS/FIMS/blob/main/inst/include/interface/rcpp/rcpp_objects/rcpp_maturity.hpp) under [inst/include/interface/rcpp/rcpp_objects](https://github.com/NOAA-FIMS/FIMS/tree/main/inst/include/interface/rcpp/rcpp_objects).
```

Comment:
issue: I do not understand the following text "maturity module is called from R using the name in quotes from the `Rcpp::class` function" and I think users could benefit from some more hand holding.

Reply:
Added:

When creating a new module in R, users can access the module using the name in quotes from the `Rcpp::class`, `LogisticMaturity`

```{r, class.source = "rchunk", eval = FALSE}
maturity <- new(fims$LogisticMaturity)
```

This module references the maturity rcpp class, `LogisticMaturityInterface` ...